src/wx/vbam-options.h

```diff
@@ -0,0 +1,272 @@
+    kDisplayFilter,
+    kDisplayFilterPlugin,
+    kDisplayIFB,
+    kDisplayKeepOnTop,
+    kDisplayMaxThreads,
+    kDisplayRenderMethod,
+    kDisplayScale,
+    kDisplayStretch,
+
+    /// GB
+    kGBBiosFile,
+    kGBColorOption,
+    kGBColorizerHack,
+    kGBLCDFilter,
+    kGBGBCBiosFile,
+    kGBPalette0,
+    kGBPalette1,
+    kGBPalette2,
+    kGBPrintAutoPage,
+    kGBPrintScreenCap,
+    kGBROMDir,
+    kGBGBCROMDir,
+
+    /// GBA
+    kGBABiosFile,
+    kGBALCDFilter,
+#ifndef NO_LINK
+    kGBALinkAuto,
+    kGBALinkFast,
+    kGBALinkHost,
+    kGBAServerIP,
+    kGBALinkPort,
+    kGBALinkProto,
+    kGBALinkTimeout,
+    kGBALinkType,
+#endif
+    kGBAROMDir,
+
+    /// General
+    kGeneralAutoLoadLastState,
+    kGeneralBatteryDir,
+    kGeneralFreezeRecent,
+    kGeneralRecordingDir,
+    kGeneralRewindInterval,
+    kGeneralScreenshotDir,
+    kGeneralStateDir,
+    kGeneralStatusBar,
+
+    /// Joypad
+    kJoypad,
+    kJoypadAutofireThrottle,
+
+    /// Keyboard
+    kJoypadDefault,
+
+    /// Keyboard
+    kKeyboard,
+
+    // Core
+    kpreferencesagbPrint,
+    kpreferencesautoFrameSkip,
+    kpreferencesautoPatch,
+    kpreferencesautoSaveLoadCheatList,
+    kpreferencesborderAutomatic,
+    kpreferencesborderOn,
+    kpreferencescaptureFormat,
+    kpreferencescheatsEnabled,
+
+#ifdef MMX
+    kpreferencesenableMMX,
+#endif
+    kpreferencesdisableStatus,
+    kpreferencesemulatorType,
+    kpreferencesflashSize,
+    kpreferencesframeSkip,
+    kpreferencesfsColorDepth,
+    kpreferencesfsFrequency,
+    kpreferencesfsHeight,
+    kpreferencesfsWidth,
+    kpreferencesgbPaletteOption,
+    kpreferencesgbPrinter,
+    kpreferencesgdbBreakOnLoad,
+    kpreferencesgdbPort,
+#ifndef NO_LINK
+    kpreferencesLinkNumPlayers,
+#endif
+    kpreferencesmaxScale,
+    kpreferencespauseWhenInactive,
+    kpreferencesrtcEnabled,
+    kpreferencessaveType,
+    kpreferencesshowSpeed,
+    kpreferencesshowSpeedTransparent,
+    kpreferencesskipBios,
+    kpreferencesskipSaveGameCheats,
+    kpreferencesskipSaveGameBattery,
+    kpreferencesthrottle,
+    kpreferencesspeedupThrottle,
+    kpreferencesspeedupFrameSkip,
+    kpreferencesspeedupThrottleFrameSkip,
+    kpreferencesuseBiosGB,
+    kpreferencesuseBiosGBA,
+    kpreferencesuseBiosGBC,
+    kpreferencesvsync,
+
+    /// Geometry
+    kgeometryfullScreen,
+    kgeometryisMaximized,
+    kgeometrywindowHeight,
+    kgeometrywindowWidth,
+    kgeometrywindowX,
+    kgeometrywindowY,
+
+    /// UI
+    kuiallowKeyboardBackgroundInput,
+    kuiallowJoystickBackgroundInput,
+    kuihideMenuBar,
+
+    /// Sound
+    kSoundAudioAPI,
+    kSoundAudioDevice,
+    kSoundBuffers,
+    kSoundEnable,
+    kSoundGBAFiltering,
+    kSoundGBAInterpolation,
+    kSoundGBDeclicking,
+    kSoundGBEcho,
+    kSoundGBEnableEffects,
+    kSoundGBStereo,
+    kSoundGBSurround,
+    kSoundQuality,
+    kSoundVolume,
+
+    // Do not add anything under here.
+    Last,
+};
```

Comment:
Could we put the string map here and add an assert for the count matching for sanity checking? I don't really like the duplication between the enum/string-map, but there is probably no better way to do that.

Reply:
I created the string map iteratively using `kAllOptionsData`, so we're no longer duplicating the data.
I still would like to find a better way of handling this down the line, we are redefining too many things for my tastes too.